airflow/hooks/S3_hook.py

```diff
@@ -82,11 +82,40 @@ def list_prefixes(self, bucket_name, prefix='', delimiter=''):
         :param delimiter: the delimiter marks key hierarchy.
         :type delimiter: str
         """
-        response = self.get_conn().list_objects_v2(Bucket=bucket_name, 
-                                                   Prefix=prefix, 
+        response = self.get_conn().list_objects_v2(Bucket=bucket_name,
+                                                   Prefix=prefix,
                                                    Delimiter=delimiter)
         return [p.Prefix for p in response['CommonPrefixes']] if response.get('CommonPrefixes') else None
 
+    def list_objects_v2(self, bucket_name, prefix='', delimiter=''):
```

Comment:
This isn't a great method name, and the docs don't make it clear when to use this over otherlist_methods.

Do we need a separate method that _just_ calls the underlying method? There's nothing to stop the callers doing `s3_hook.get_conn().list_objects_v2` directly.

Reply:
The original reasoning for this here was for use in conjunction with an operator that I have, which utilize this method directly, in which case having it here and well-tested would be important to ensure sustainability of the operator. Since the operator is out of the scope of the PR, perhaps I should remove the method & test here and reserve it for a later PR if it's usage is more clear?

Reply:
My comment was mainly around the function name, specifically the `_v2` suffix. That and the docs don't make it clear the difference between this and prefixes, keys.

It is probably a useful addition still.